library/core/src/panic.rs

```diff
@@ -140,6 +140,36 @@ pub macro unreachable_2021 {
     ),
 }
 
+/// Invokes a closure, aborting if the closure unwinds.
+///
+/// When compiled with aborting panics, this function is effectively a no-op.
+/// With unwinding panics, an unwind results in another call into the panic
+/// hook followed by a process abort.
+///
+/// # Notes
+///
+/// Instead of using this function, code should attempt to support unwinding.
+/// Implementing [`Drop`] allows you to restore invariants uniformly in both
+/// return and unwind paths.
+///
+/// If an unwind can lead to logical issues but not soundness issues, you
+/// should allow the unwind. Opting out of [`UnwindSafe`] indicates to your
+/// consumers that they need to consider correctness in the face of unwinds.
```

Comment:
does this makes sense, given that there seems to be some sentiment around deprecating the traits?

Reply:
I wrote this in since the traits are still in use. Reality may be that `UnwindSafe` gets ignored without much thought since it often just goes unimplemented due to overlooking it, instead of indicating a potential failure point, but I do think it's okay to be a bit aspirational here, bringing it up as an option that's still better than aborting.